mathics_scanner/tokeniser.py

```diff
@@ -326,28 +337,54 @@ def __repr__(self):
 
 
 class Tokeniser(object):
+    """
+    A tokeniser for the Wolfram Language.
+
+    When subclassing ``Tokeniser``, custom tokenisation rules can be defined by 
```

Comment:
I don't think this mechanism should be exposed the public API (and therefore it shouldn't be documented in the docstring). If you think about it, all consumers of this library want is to have a functioning WL tokeniser that they can use as a black-box (that what I think at-least).

This definitively should be documented somewhere though. @rocky I'd apprciate if we could merge https://github.com/Mathics3/mathics-scanner/pull/8 before this, so that I can move this information to `implementation.rst`. I also plan to convert `implementation.rst` and the rest of the documentation to a proper Sphinx document before we release the library (we should be pretty easy to do, so it's not gonna take too much time).